log/zerolog.go

```diff
@@ -0,0 +1,139 @@
+/*
+ *    Copyright 2019 Insolar
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package log
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+
+	"github.com/insolar/insolar/configuration"
+	"github.com/insolar/insolar/core"
+)
+
+type zerologAdapter struct {
+	logger   zerolog.Logger
+}
+
+func newZerologAdapter(cfg configuration.Log) (*zerologAdapter, error) {
+	var output io.Writer
+	switch strings.ToLower(cfg.Formatter) {
+	case "text":
+		output = zerolog.ConsoleWriter{Out: os.Stderr, NoColor: true, TimeFormat: timestampFormat}
+	case "json":
+		output = os.Stderr
+	default:
+		return nil, errors.New("unknown formatter " + cfg.Formatter)
+	}
+
+	zerolog.CallerSkipFrameCount = 4
```

Comment:
Could you add some comment with description why 4 or link on doc where this description present?

Reply:
no doc, no description, added test to show that it is correct